ueberzug/X/display.c

```diff
@@ -92,25 +93,50 @@ has_property(DisplayObject *self, Window window, Atom property) {
     return status == Success && !(actual_type_return == None && actual_format_return == 0);
 }
 
+static void
+get_child_window_ids_helper(Display *display, Window window,
+        Window **total_children, unsigned int *total_num_children) {
+    Window _, *children;
+    unsigned int num_children;
+
+    XQueryTree(display, window, &_, &_, &children, &num_children);
+    if (!children) return;
+
+    if (!*total_children) {
+        *total_children = calloc(num_children, sizeof(Window));
+    } else {
+        Window *tmp = reallocarray(*total_children, num_children + *total_num_children, sizeof(Window));
```

Comment:
If `realloc` fails, won't this go out of bound later down? Also `reallocarray` is a non-standard BSD extension. Don't know what the policy is on the ueuberzug codebase towards using such extensions.

Reply:
>Also reallocarray is a non-standard BSD extension.

is it?

```
reallocarray():
           Since glibc 2.29:
               _DEFAULT_SOURCE
           Glibc 2.28 and earlier:
               _GNU_SOURCE
```

Reply:
Those are just feature test macros. See the CONFORMING TO section:

```
CONFORMING TO
       malloc(), free(), calloc(), realloc(): POSIX.1-2001, POSIX.1-2008, C89, C99.

       reallocarray() is a nonstandard extension that first appeared in OpenBSD 5.6 and FreeBSD 11.0.
```

Reply:
I see, currently there is no explicit C standard enforced in the code (gcc's default perhaps?).